astropy/io/fits/tests/test_convenience.py

```diff
@@ -23,3 +26,13 @@ def test_resource_warning(self):
         with catch_warnings() as w:
             header = fits.getheader(self.data('test0.fits'))
             assert len(w) == 0
+
+    def test_table_to_hdu(self, tmpdir):
+        data = np.array(
+            list(zip([1, 2, 3], ['a', 'b', 'c'], [2.3, 4.5, 6.7])),
+            dtype=[(str('a'), int), (str('b'), str('U1')), (str('c'), float)])
+        table = Table(data)
```

Comment:
The "astropy" way of doing this (IMHO) is cleaner and more readable:
```
table = Table([[1, 2, 3], ['a', 'b', 'c'], [2.3, 4.5, 6.7]], 
              names=['a', 'b', 'c'], dtype=['i', 'U1', 'f'])
```
And note that `Table` allows unicode for `names` and `dtype` so no `str()` casting needed.